src/Sarif.Sarifer/TextBufferViewTracker.cs

```diff
@@ -61,10 +61,23 @@ public void RemoveTextView(ITextView textView)
             if (trackingInformation.Views.Count == 0)
             {
                 this.bufferToViewsDictionary.Remove(textView.TextBuffer);
+                trackingInformation.CancellationTokenSource.Dispose();
                 LastViewRemoved?.Invoke(
                     this,
-                    new LastViewRemovedEventArgs(trackingInformation.LogId));
+                    new LastViewRemovedEventArgs(trackingInformation.Path));
             }
         }
+
+        /// <inheritdoc/>
+        public void Clear()
```

Comment:
>Clear [](start = 20, length = 5)

Who calls this?

Reply:
When we dispose the class `BackgroundAnalysisTextViewCreationListener`

---
In reply to: [539663388](https://github.com/microsoft/sarif-visualstudio-extension/pull/312#discussion_r539663388) [](ancestors = 539663388)